core/lib/multistore.ts

```diff
@@ -77,18 +78,42 @@ export function prepareStoreView (storeCode: string) : StoreView {
     initializeSyncTaskStorage()
     Vue.prototype.$db.currentStoreCode = storeView.storeCode
   }
+
   return storeView
 }
 
 export function storeCodeFromRoute (matchedRouteOrUrl: LocalizedRoute | RawLocation | string) : string {
```

Comment:
would you mind to add unit tests for this helper?
It would be great for future changes here to not miss any part, and it's pretty critical functionality :)

Reply:
sure, added